src/symbols_linux.cpp

```diff
@@ -394,17 +401,19 @@ void ElfParser::parseDynamicSection() {
                     _cc->addImport((void**)(base + r->r_offset), strtab + sym->st_name);
                 }
             }
-        } else if (rel != NULL && relsz != 0) {
+        }
+
+        if (rel != NULL && relsz != 0) {
             // Shared library was built without PLT (-fno-plt)
             // Relocation entries have been moved from .rela.plt to .rela.dyn
             for (size_t offs = relcount * relent; offs < relsz; offs += relent) {
                 ElfRelocation* r = (ElfRelocation*)(rel + offs);
-                if (ELF_R_TYPE(r->r_info) == R_GLOB_DAT) {
+                if (ELF_R_TYPE(r->r_info) == R_GLOB_DAT || ELF_R_TYPE(r->r_info) == R_DIRECT) {
                     ElfSymbol* sym = (ElfSymbol*)(symtab + ELF_R_SYM(r->r_info) * syment);
                     if (sym->st_name != 0) {
-                        _cc->addImport((void**)(base + r->r_offset), strtab + sym->st_name);
+                        _cc->tryAddImport((void**)(base + r->r_offset), strtab + sym->st_name);
```

Comment:
Maybe just switch the order of these two loops? This will ensure proper precedence automatically without override logic in CodeCache?